### PR TITLE
add Catch2 C++ extension

### LINKED ISSUE #366: Request: Add Catch2 C++ unit test support
Catch2 C++ unit test framework: https://github.com/catchorg/Catch2

Catch2 tests can be given an argument to specify different "reporters" to change the output format, as discussed here: https://github.com/catchorg/Catch2/blob/master/docs/reporters.md

It defaults to console, in human readable form.  But, you can select others such as `junit` which is XML corresponding to Ant's junitreport target, or `xml` which I'm thinking is the best and is XML tailored to Catch.

There's also Visual Studio Code extension that adapts Catch2 and Google Test to VSCode here: https://github.com/matepek/vscode-catch2-test-adapter -- perhaps seeing this will help with my request and #294.  A very brief look at their code makes it look like they just let them run reporting to the console, and parse that output.

### PR DESCRIPTION
**Pull request by bbli**
Make sure these boxes are checked before submitting your pull request:

- [x] Add fixtures and spec when implementing or updating a test runner
- [x] Update the README accordingly
- [x] Update the Vim documentation in `doc/test.txt`

This should help resolve #366

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ runners are supported:
 |       Language | Test Runners                                                                                                       | Identifiers                                                                                                                                  |
 | -------------: | :----------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------- |
 |         **C#** | .NET                                                                                                               | `xunit`, `dotnettest`                                                                                                                        |
+|         **C++** | ctest                                                                                                               | `ctest`, `dotnettest`                                                                                                                        |
```
Comment (codeinabox) [inline README.md:25]: Should `dotnettest` be part of this or is this a typo?

Comment (bbli) [reply README.md]: Typo, thanks for the catch! I have updated the README with the correct info

File: README.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ runners are supported:
 |       Language | Test Runners                                                                                                       | Identifiers                                                                                                                                  |
 | -------------: | :----------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------- |
 |         **C#** | .NET                                                                                                               | `xunit`, `dotnettest`                                                                                                                        |
+|         **C++** | CTest,Make                                                                                                               | `ctest`, `make`                                                                                                                        |
```
Comment (codeinabox) [inline README.md:25]: This just adds support for Catch2 right? If so you want to enter 

```md
**C++** | Catch2  | `catch2`
```

**Review by codeinabox — CHANGES_REQUESTED** [review]
LGTM just some small feedback about the README and could you please also update `doc/test.txt`

File: README.md (hunk 2 above)
Comment (bbli) [reply README.md:25]: Hmm, so the issue is that in C++, the testing framework doesn't come with a test runner, and instead you have to build it. One implication of this is that the test runner's name is whatever the target's name is in the build system of choice(which I defaulted to `make`), so you can't call `catch2` on the command line like you can with say `pytest`, or `cargo test`.

Let me know what you think the best way is to make this info clear to future readers!

Comment (codeinabox) [reply README.md:25]: [Every runner should add a new Vim command for invoking it](https://github.com/vim-test/vim-test#runner-commands). If you first run `let g:test#runner_commands = ['Catch2']` are you able to invoke the command?

**Comment by codeinabox** [comment]
Hello there, just checking if you'd like a hand with this? I could check out your branch and determine what the runner command is meant to b

**Comment by bbli** [comment]
> Hello there, just checking if you'd like a hand with this? I could check out your branch and determine what the runner command is meant to b

Ohh wow, thanks for the reminder. I sorta want to try to figure this out by myself. Can you let me work on it this weekend and get back to you afterwards?

File: README.md (hunk 2 above)
Comment (bbli) [reply README.md:25]: Hi @codeinabox , looks like I was overthinking it, as adding that does indeed allow me to invoke the test runner from within Vim, which makes sense as from reading the code it calls the same `test#executable` function as the general vim-test commands do. I added the command runner's usage in the recent commit below. Can you review?

**Review by codeinabox — APPROVED** [review]
Thank you for taking the time with this, it's very much appreciated and congratulations on being the first 🥇 to add a C++ test runner

**Comment by bbli** [comment]
> Thank you for taking the time with this, it's very much appreciated and congratulations on being the first 1st_place_medal to add a C++ test runner

Thanks @codeinabox, it was my pleasure, as I can now use vim-test to test drive C++ code!

**Pull request merged**